Description/PositionEnhancer.php

```diff
@@ -0,0 +1,79 @@
+<?php
+
+namespace Sonata\DoctrinePHPCRAdminBundle\Description;
+
+use Doctrine\Common\Persistence\ManagerRegistry;
+use PHPCR\PathNotFoundException;
+use PHPCR\Util\PathHelper;
+use Symfony\Cmf\Component\Resource\Description\Description;
+use Symfony\Cmf\Component\Resource\Description\DescriptionEnhancerInterface;
+use Symfony\Cmf\Component\Resource\Puli\Api\PuliResource;
+use Symfony\Cmf\Component\Resource\Repository\Resource\CmfResource;
+
+/**
+ * A description enhancer to add the position/sorting value of children on a parent.
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@mayflower.de>
+ */
+class PositionEnhancer implements DescriptionEnhancerInterface
+{
+    /**
+     * @var \PHPCR\SessionInterface
+     */
+    private $session;
+
+    public function __construct(ManagerRegistry $manager, $sessionName)
+    {
+        $this->session = $manager->getConnection($sessionName);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function enhance(Description $description)
+    {
+
+        $nodePath = $description->getResource()->getPath();
+        $nodeName = PathHelper::getNodeName($nodePath);
+        $parentPath = PathHelper::getParentPath($nodePath);
+
+        try {
+            $parentNode = $this->session->getNode($parentPath);
+        } catch (PathNotFoundException $exception) {
+            return false;
+        }
+
+        $nodeIterator = $parentNode->getNodes();
+        $nodeIterator->rewind();
+        $counter = 0;
+        while ($nodeIterator->valid()) {
+            $counter++;
+            if ($nodeIterator->key() === $nodeName) {
+                break;
+            }
+            $nodeIterator->next();
+        }
+
+        $description->set('position', $counter);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function supports(PuliResource $resource)
+    {
+        if (!$resource instanceof CmfResource) {
+            return false;
+        }
+
+        try {
+            $parentNode = $this->session->getNode(PathHelper::getParentPath($resource->getPath()));
+        } catch (PathNotFoundException $exception) {
+            return false;
+        }
+
+        // Todo: check for non orderable type
```

Comment:
@dbu @lsmith77 @dantleech  do we have that orderable Type on a phpcr node? (https://docs.adobe.com/content/docs/en/spec/jcr/2.0/23_Orderable_Child_Nodes.html)